prow/plugins/trigger/generic-comment.go

```diff
@@ -55,7 +55,10 @@ func handleGenericComment(c Client, trigger plugins.Trigger, gc github.GenericCo
 
 	presubmits, err := c.Config.GetPresubmits(c.GitClient, org+"/"+repo, refGetter.BaseSHA, refGetter.HeadSHA)
 	if err != nil {
-		return fmt.Errorf("failed to get presubmits: %v", err)
```

Comment:
This seems pretty large in functionality change -- is it reasonable to test this flow?

Reply:
You are right in that its not completely intuitive at first why its needed and its super important, so I have refactored the code a bit and then added tests